compiler/rustc_ast/src/visit.rs

```diff
@@ -65,7 +65,15 @@ impl BoundKind {
 #[derive(Copy, Clone, Debug)]
 pub enum FnKind<'a> {
     /// E.g., `fn foo()`, `fn foo(&self)`, or `extern "Abi" fn foo()`.
-    Fn(FnCtxt, &'a Ident, &'a FnSig, &'a Visibility, &'a Generics, &'a Option<P<Block>>),
+    Fn(
+        FnCtxt,
+        &'a Ident,
+        &'a FnSig,
+        &'a Visibility,
+        &'a Generics,
+        &'a Option<P<FnContract>>,
+        &'a Option<P<Block>>,
+    ),
```

Comment:
I think what could be useful on master is to turn all of these individual fields into a single `&'a Fn`, then there should be no diff in this file beyond the new contract methods